File: app/src/main/java/com/example/steaminvestmentmanager/utilclasses/ItemListUpdater.java
package com.example.steaminvestmentmanager.utilclasses;

import com.example.steaminvestmentmanager.MainActivity;

public class ItemListUpdater {
    private static MainActivity mainActivity;

    public static void setMainActivity(MainActivity mA) {
        mainActivity = mA;
    }

    public static MainActivity getMainActivity() {
        return mainActivity;
    }

    public static void updateSteamItemAdapter() {
        mainActivity.runOnUiThread(() -> MainActivity.steamItemAdapter.notifyDataSetChanged());
    }
}

File: app/src/main/java/com/example/steaminvestmentmanager/utilclasses/ItemListUpdater.kt
package com.example.steaminvestmentmanager.utilclasses

import com.example.steaminvestmentmanager.MainActivity

object ItemListUpdater {
    var mainActivity: MainActivity? = null
    fun updateSteamItemAdapter() {
        mainActivity!!.runOnUiThread { MainActivity.steamItemAdapter!!.notifyDataSetChanged() }
    }
}